src/libostree/ostree-repo.c

```diff
@@ -5105,7 +5107,26 @@ _ostree_repo_gpg_verify_data_internal (OstreeRepo    *self,
         return NULL;
 
       if (gpgkeypath)
-        _ostree_gpg_verifier_add_key_ascii_file (verifier, gpgkeypath);
+        {
+          if (strchr (gpgkeypath, ','))
```

Comment:
Is there any reason you are going for these two different code paths, rather than just always doing the `g_strsplit` and loop, since I would assume that running the split on a string without `,` would result in a vector with one entry?

Reply:
At first I was following from @cgwalters comment on the issue https://github.com/ostreedev/ostree/issues/773#issue-218950903 that a check for the case of a single string should be done first, before interpreting as multiple due to the trailing `;` being required.

Looks like that requirement has changed though since then. In https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-1.1.html#value-types the trailing `;` is now optional, it used to be required in 1.0.

I see the way I've done it though with `split_string`, the trailing separator is not properly handled anyway as an empty string would be added at the end.

`g_key_file_get_string_list` should handle the single case and any trailing `,` properly. I'll add this in, have this always be called and loop. Will also add a test to make sure the trailing `,` is handled.

Thanks for pointing that out!